crates/core/tedge_watchdog/src/systemd_watchdog.rs

```diff
@@ -33,6 +33,13 @@ use tracing::warn;
 
 const SERVICE_NAME: &str = "tedge-watchdog";
 
+/// How many times more often do we send notify to systemd watchdog, than is necessary from the
+/// `WatchdogSec` value.
```

Comment:
It couldn't be `WatchdogSec` value itself, as there's the risk of missing notifications due to the additional delay between the health check request and response. For e.g, assume a `WatchdogSec` value of `30s` and the systemd watchdog timer starts at `T`. If our tedge_watchdog wakes up from its sleep at `T+29` and sends the health check request, the response might be slightly delayed and received at `T+31` leading to the watchdog timer expiring leading to a restart.

That's why we started with half of `WatchdogSec` which gives sufficient buffer for the responses to arrive. Theoretically that should have worked, but in practice, even that interval seemed insufficient and we still encountered occasional notification misses, which lead to the quarter value.

BTW, even smaller fractions won't work if the `WatchdogSec` value itself is too small, like a couple of seconds, as the resultant computed delay of 1 sec might also not work, if the request-response delay is higher.